### ISSUE TITLE
Future Development of Homebridge Hue

### ISSUE THREAD

**Issue by ebaauw** [ebaauw/homebridge-hue#1070]
Homebridge Hue is my oldest plugin.  I started working on it in February 2016, and published the first version on GitHub in October 2016.  I added support for deCONZ in the summer of 2017, including web socket notifications.  In February 2020, I added HTTPS support for the gen-2 Hue bridge, followed by API v2 push notifications in December 2021.
Over that period, the number of different devices supported by Homebridge Hue has increased dramatically, fuelled by the consumerisation of Zigbee devices.  Starting with (then) Philips Hue, OSRAM Lightify, and IKEA Trådfri, followed by Xiaomi, and many OEM devices, and even devices from Lidl and Aldi.
The code base has become increasingly complex over the years, and is due for a refactor.  Homebridge Hue still uses Homebridge's traditional static platform accessory model (see #4), where most recent plugins use the dynamic platform accessory model, which offers many benefits.  The Hue and deCONZ APIs have been diverging, and their v2 APIs are completely different, making it increasingly hard to combine Hue bridge and deCONZ gateway support in a single plugin.

Homebridge Hue is also my most used plugin.  I don't collect any statistics, but the NPM registry mentions up to 8,000 weekly downloads, and GitHub Insights lists over 3,400 unique visitors.
Homebridge Hue is also, by far, my largest plugin; I'm exposing over 140 accessories from my deCONZ gateway.  I would take me half a day to reconfigure HomeKit if those accessories would be removed from HomeKit and re-exposed.
Because of this, I've been hesitant to refactor Homebridge Hue.  While I can technically manage to expose my accessories using a dynamic platform plugin, without HomeKit losing them, this is tricky and requires a deep understanding of the internal working of Homebridge and HAP-NodeJS.  For me, a big-bang update would be out of the question, as I think for most users.

Going forward, I will create a new plugin for deCONZ, and a new plugin for Hue APIv2, while keeping the current Homebridge Hue plugin.  These three plugins can be run in parallel, using different child bridges, connecting to the same Hue bridge(s) and/or deCONZ gateway(s).  This allows you to setup a new HomeKit home for testing.  By connecting the child bridges for all plugins to the same Home, white- and blacklisting can be used to move accessories over to the new plugins one-by-one.  This will also allow me to release work-in-progress versions of the new plugins, not yet supporting all device types.
Eventually the current Homebridge Hue plugin will drop support for the gen-2 Hue bridge and the deCONZ gateway.  After that, I might release a new version with the dynamic platform accessory model.  I don't think there are enough users left using the gen-1 Hue bridge or HA-Bridge or Tasmote to justify the effort needed to provide a gradual migration path.

**Comment by HARDIK-TSH1392** [comment]
Hey, I am new to open source and I am finding it really difficult as to how to start doing anything. I just came across your project, could you guide/explain as to what should I do because I am interested in contributing to open source, so taking your project as a start can you help me contribute?

**Comment by ebaauw** [comment]
@HARDIK-TSH1392 better start with something far less technically complex.

**Comment by sschuste** [comment]
Wow. That's a lot of work you have to do. So much code and I'm only using it for two devices. And I don't have enough programming skills to contribute anything essential. If you're ever in Frankfurt am Main, your dinner's on me. But maybe you have some better ideas how we can support you.

**Comment by vvdheuvel** [comment]
Hi Erik, wise decision! 🙏🏻 I think this will take out some complexity and better focus on both platforms. When do you suspect it will stable for usage? Will there be a possibility to migrate?

**Comment by jan666** [comment]
Oh, that sounds really horrible.

I guess its safe to just use the old homebridge-hue plugin forever? Re-doing everyhing in the Home App will be a real pain

**Comment by ebaauw** [comment]
Yes, that is horrible.  That's why I had been delaying the refactor of Homebridge Hue for so long, up to the point where it's no longer sustainable to maintain it.

I cannot prevent that pain, unfortunately, but I won't force you when to take it, nor to take all of it at once.  I will stop adding new devices and features in Homebridge Hue pretty soon, after both Homebridge deCONZ and Homebridge Hue2 are ready for mainstream production.  I will eventually release a version 1.0 of Homebridge Hue that no longer supports the deCONZ gateway nor the gen-2 Hue bridge, but not within the next six months.  Of course, you can continue to run (and even re-install) the 0.x version of Homebridge Hue.

**Comment by Ar-eh** [comment]
Does using an app like "Controller for HomeKit", with the ability to backup a HomeKit configuration and restore from that backup, address the migration difficulties? I've found that when migrating between Homebridge Hue and the HomeKit integration in HomeAssistant, the backup does not work perfectly due to differences in how accessories are exposed. But perhaps it would be smoother in migrating from Homebridge Hue to HomeBridge deCONZ?

**Comment by ebaauw** [comment]
I really don’t know.

**Comment by jan666** [comment]
@ebaauw can you provide a "migration guide"?

**Comment by ebaauw** [comment]
As I said before: I really don’t know.  I have no experience with Controller for HomeKit.

I’m currently running Homebridge deCONZ in a test home.  When I think it’ll be ready to replace Homebridge Hue in my real home, I’ll release v1.0 including a WiKi on migration.

**Comment by vvdheuvel** [comment]
Let me help you out a little bit Erik, first of all it's best practice to always back-up your home automation settings and configurations. Hardware can die or corrupt (same goes for the software). Controller helps you out a bit on this regarding HomeKit. It backs-up your HomeKit settings and HomeKit codes and leverages advanced HomeKit automations. You should also back-up your HomeBridge set-up/bridges before upgrading, in case you'll have to do a disaster recovery/roll-back if anything goes to 💩. So OS update or plug-in update ➡︎ back-up first! Hope this helps a bit? So short version: yeah it will help you out!

**Comment by ebaauw** [comment]
Please refrain from advertising other products in this repo.

**Comment by vvdheuvel** [comment]
If you’re referring to Homebridge/Controller? Just trying to help out! So also no more mentions of deCONZ, Hue?

**Comment by mhin76** [comment]
So much appreciate your work and commitment!! It is the most important plugin on my homebridge for about 30 devices. Good luck for the refactoring and thanks for your dedication!

**Comment by jaydisc** [comment]
Controller4HomeKit would handle this transition perfectly. I moved all my accessories from a Hue Bridge to deCONZ a year or so ago... after doing so, I matched things up in Controller, and it rebuilt all room assignments, scenes and automations in seconds. It's quite amazing, and imo, removes any worry or concern about migrating between homebridge-hue and homebridge-deconz, etc.

**Comment by beatpaul** [comment]
Hi Erik!
Im a real fan of homebridge hue and u helped me out e few times to keep it run flawless :)

But now i have an issue that worries me:
I installed homebridge deconz (separate bridge) and now my switches are responding (blinking in home app when pressed) but the  assigned accessories are not switched… when i press „Diese Aktion testen“ ist works thou
Any idea ?

Thanx 
Paul

**Comment by bercie** [comment]
hue app/homebridge verwijst naar v16.14.0, kan deze nergens vinden?

**Comment by krysly** [comment]
I'm just getting into the topic of homebridge and Hue and would like to use the new v2 version directly for this.
The v2 must currently be configured in the config file. However, there is currently no wiki or example for this as it exists in the first version in github.
Would it be possible to provide a basic configuration for v2 as an example ? 
I tried a v1 config but that doesn't work.
Many thanks
Christian

**Comment by ebaauw** [comment]
Homebridge Hue2 doesn't yet do anything.

**Comment by ghdbwise** [comment]
Hi Eric,
does homebridge-deconz do anything? I see a stateless switch under devices and in homekit. When restarting homebridge there is no pairing request. The deconz is in pairing mode, but nothing happens.
Many thanks so far for homebridge-hue.
Regards Peter.

**Comment by ebaauw** [comment]
Yes, it’s still under development, but functional.  You need to select what to expose in Eve or another HomeKit app, see the README.

**Comment by BorussiaPlayer17** [comment]
Well I'm using the hue emulation on a Tasmota device. :-(

**Comment by Jacksonbm1** [comment]
Just saw this in the logs. Thanks for letting us know. Refactors don't get rewarded the way they should. They are better for everyone in the long run, but they often look like an inconvenience to the user. Props for doing it.

**Comment by tomsteenbakkers** [comment]
Eric, If you need any help on testen homebridge-hue2, please drop me a note! I happy to help, I really appreciate that you take the time to refactor. This is the only way to keep legacy out of the Homebridge domain.

**Comment by iMonZ** [comment]
> Wow. That's a lot of work you have to do. So much code and I'm only using it for two devices. And I don't have enough programming skills to contribute anything essential. If you're ever in Frankfurt am Main, your dinner's on me. But maybe you have some better ideas how we can support you.

Same for me in Berlin

**Comment by TecTolik** [comment]
Hello Eric,
thank you very much for you developement! I have tested it in my szenario with the Homebridge deConz plugin. After i figured it out how to expose everything in "eve", it worked fine. But in my Homebridge Log im getting every minute this entry. Can you tell me why it getting throwed all the time ? 
![image](https://user-images.githubusercontent.com/43959471/169498766-699572f9-5e4a-406f-be7f-00332cfa74d1.png)

Thank you!

**Comment by ebaauw** [comment]
@TecTolik see https://github.com/ebaauw/homebridge-deconz/issues/30

**Comment by Ar-eh** [comment]
> Does using an app like "Controller for HomeKit", with the ability to backup a HomeKit configuration and restore from that backup, address the migration difficulties? I've found that when migrating between Homebridge Hue and the HomeKit integration in HomeAssistant, the backup does not work perfectly due to differences in how accessories are exposed. But perhaps it would be smoother in migrating from Homebridge Hue to HomeBridge deCONZ?

I migrated from Homebridge Hue to Homebridge Deconz today, and used Controller to create a backup before the migration and restore from the backup after migration. It was pretty painless and the backup restore worked out pretty well (thankfully, since I have 167 accessories). The only accessories that did not restore correctly were my Aqara wall switches, but deleting the service from the backup and reconfiguring those wasn't a big deal. The only thing that did not restore from backup were the custom icons for all of the accessories, but again, not a big deal. I'll post again if I notice any other issues arising from the migration. Thank you for all of your work on this, Erik!

**Comment by pponce** [comment]
@Ar-eh Any chance you can do a short write up on how to used Controller to do the migration?
Controller is an expensive app. Likely this migration alone will justify the cost if it's as easy as it seems to have gone for you.
I use homebridge Hue to expose of of my Hue Hub connected devices to homekit and also use it to expose my Deconz devices to homekit.

**Comment by Eddy-the-Git** [comment]
I manage *a lot* of Deconz devices via homebridge-hue and don't want my life to suck when upgrading to homebridge-deconz. Would this work?

1) Block access to port 51692 on the homebridge machine via a firewall (mine listens on that port for Homekit communication). I think this would make Homekit show "No response" for all homebridge devices

2) In homegridge: Remove homebridge-hue, then install homebridge-deconz and make sure all devices are available

3) Unblock access to port 51692 on the homebridge machine

**Comment by ebaauw** [comment]
I'm not sure what you're trying to accomplish with that.  The accessories as exposed by Homebridge deCONZ would probably still get new `aid`s, causing Homebridge to see them as new accessories.

**Comment by ikorbln** [comment]
Hello Erik, is there a timeline when the good old hue plugin stop supporting deconz? I want to be prepared for a migration because when the homebridge stop working my wife is not excited ;-)
And in this future, i will end up with two plugins? one for the hue and one for the deconz? and they can co-exists on the same homebridge?

**Comment by ebaauw** [comment]
The plugin won’t just stop working, but I’ll stop adding support for new devices and features, and eventually I might publish a version that no longer supports deCONZ.

Yes, there will be two (new) plugins, one for deCONZ and one for the 2nd-gen Hue bridge (using the Hue API v2).  They will co-exist in the same Homebridge instance, but I would recommend to configure child bridges for each plugin anyways.

I’m sorry, no timeline.  I’m doing this in my spare time.  I want to make sure Homebridge deCONZ provides all Homebridge Hue features, before deprecating deCONZ support in Homebridge Hue.

**Comment by murtazabasrai** [comment]
Hi @ebaauw, I am doing a fresh setup of homebridge with hue. The sole intention for using the homebridge-hue plugin is for exposing the tap-dial switch "long-press" events to homekit.
I am not sure if I should start using the new (homebridge-hue2)[https://github.com/ebaauw/homebridge-hue2] plugin for this or I still need to use this old plugin?

Also thank you for your efforts! I went through this thread and the plan look promising.

**Comment by ebaauw** [comment]
Homebridge Hue2 is still a placeholder at the moment, so for the Hue bridge, use Homebridge Hue.

**Comment by The-Paw** [comment]
Hello .. got this error on hoobs. ... 

12.7.2023, 20:26:41HOOBSHueWARNUNG192.168.178.47: request 1: connect ECONNREFUSED 192.168.178.47:1881
12.7.2023, 20:26:41HOOBSHueFEHLER192.168.178.47:1881: not found - retrying in 30s

What could it be ??

**Comment by martind1983** [comment]
> Hello .. got this error on hoobs. ...
> 
> 12.7.2023, 20:26:41HOOBSHueWARNUNG192.168.178.47: request 1: connect ECONNREFUSED 192.168.178.47:1881 12.7.2023, 20:26:41HOOBSHueFEHLER192.168.178.47:1881: not found - retrying in 30s
> 
> What could it be ??

you're hijacking a thread, but I have the same Issue and don't know the solution to fix the root cause. I checked with "systemctl status deconz" all running and on the right port.
I could temporarily solve it with "sudo systemctl restart deconz" - ... until the next restart of homebridge, same issue. Maybe this helsp you as well for now....

**Comment by protons3th** [comment]
I've been using Hue v1 and am now looking to integrate v2 to future-proof and test its capabilities.

I'm not sure how to add a child bridge manually or find documentation as per this message: "This plugin must be configured manually using the Homebridge UI Config Editor. Please consult the plugin documentation for instructions on how to correctly configure this plugin."

I must say, I'm quite pleased with the Homebridge plugin so far. Thousands of users, including myself, owe you a big thank you for your work.

**Comment by ebaauw** [comment]
Homebridge Hue2 is still just a placeholder at the moment, so for the Hue bridge, use Homebridge Hue.

> I'm not sure how to add a child bridge manually 

See the Homebridge WikI: https://github.com/homebridge/homebridge/wiki/Child-Bridges.

> or find documentation as per this message: "This plugin must be configured manually using the Homebridge UI Config Editor. Please consult the plugin documentation for instructions on how to correctly configure this plugin."

Not sure where that messages comes from, but it's probably the result of Homebridge Hue2 not yet containing a `config.schema.json`, which defines the _Settings_ form to the Homebridge UI.

**Comment by protons3th** [comment]
> Homebridge Hue2 is still just a placeholder at the moment, so for the Hue bridge, use Homebridge Hue.

Ok, I'll just let it stay under my plugins as disabled then.

> Not sure where that messages comes from, but it's probably the result of Homebridge Hue2 not yet containing a `config.schema.json`, which defines the _Settings_ form to the Homebridge UI.

It happens whenever I click settings or bridge settings in Homebridge plugins.

Thank you for the fast reply

**Comment by ebaauw** [comment]
**UPDATE AUG 2023**
I've bumped Homebridge deCONZ to v1, as I think it's production ready.  I concluded my migration to Homebridge deCONZ in May, and have been running Homebridge deCONZ exclusively for three months now, without any issues.

Consequently, Homebridge Hue support for deCONZ gateways has been deprecated, meaning you should start planning a migration to Homebridge deCONZ.  For now, Homebridge Hue continues to expose devices connected to a deCONZ gateway, but I won't be adding support for new devices nor implement any feature requests not supported by the Hue bridge.

Note that Homebridge Hue2 is still just a placeholder, so continue to use Homebridge Hue to expose devices connected to a Hue bridge.

**Comment by ebaauw** [comment]
### Familiarise yourself with Homebridge deCONZ
1. Install Homebridge deCONZ in a child bridge.  See Homebridge deCONZ README for details, incl. link how to setup a child bridge;
2. Pair Homebridge deCONZ to your deCONZ gateway; by default, it will expose all zigbee devices connected to the deCONZ gateway.
3. Temporarily create a new HomeKit Home, and pair the Homebridge deCONZ child bridge to that Home;
4. Check that your devices are exposed correctly.  Log an issue for devices that aren't exposed correctly or for which you need additional feature requests.
5. Play around with the dynamic configuration through the `ui` command-line tool, un-exposing and exposing devices, changing log level, how to expose devices, etc.

### Move Homebridge deCONZ to production
7. Un-expose all devices in Homebridge deCONZ;
8. Remove Homebridge deCONZ from your test HomeKit home, and pair it to your real production HomeKit home.  Remove the test HomeKit home;
9. Expose a device (that passes the checks in step 4) in Homebridge deCONZ.  Note the message of the migration resource link in the Homebridge log.
You'll now see the accessory twice in HomeKit, once exposed by Homebridge Hue, and once by Homebridge deCONZ.
10. Configure the Homebridge deCONZ accessory in HomeKit: assigning it to the HomeKit rooms, groups, scenes, automations like the Homebridge Hue exposed accessory.
11. Update the Homebridge Hue config.json setting `homebridgeDeconz` to the ID of the migration resourcelink by Homebridge deCONZ from step 8.
12. Restart Homebridge Hue to apply the migration resourcelink, blacklisting the devices exposed by Homebridge deCONZ.
13. At your convenience, repeat steps 8, 9, 11 for each device connected to the deCONZ gateway.

### Cleanup Homebridge Hue
14. After all devices have been migrated, remove the deCONZ gateway from the Homebridge Hue configuration.
15. Uninstall Homebridge Hue once all gateways are migrated (unless you still use it for any Hue bridge).

**Comment by kolorfilm** [comment]
Thank you @ebaauw for the very detailed migration guide! 🙏🏻

I‘m a little bit confused about step 7
`Un-expose all devices in Homebridge deCONZ`

How can I do this? Also when I add the gateway it automatically expose all devices right? How can I prevent this to only expose one by one as you described starting from step 9?

What does this „migration resource link“ thing? I do not completely understand it. I mean I could also just add the new gateway, remove the old one from HomeKit and resetup everything?

How does the „migration resource link“ help here?

Thank you already for your help!

**Comment by ebaauw** [comment]
> I‘m a little bit confused about step 7
> > Un-expose all devices in Homebridge deCONZ
> How can I do this?

Issue `ui put /accessories/`_id_` '{"expose": false}'` for each accessory.  Hence: familiarise yourself with Homebridge deCONZ, notably step 5.

> What does this „migration resource link“ thing?

It lists the resources exposed by Homebridge deCONZ, so Homebridge Hue can stop exposing these.

> I mean I could also just add the new gateway, remove the old one from HomeKit and resetup everything?

Yes, you could, if you want to do a big-bang migration.  Depending on the size of your HomeKit home, this might take you a full day or longer.

> How does the „migration resource link“ help here? And also when does it show up and where?

It's a `/resourcelinks` resource on the deCONZ gateway, maintained by Homebridge deCONZ, and read by Homebridge Hue.  Homebridge deCONZ writes a log messsage, whenever the exposed devices change, or when issuing _Identify_ on the gateway accessory.
```
[8/12/2023, 11:30:07 AM] [deCONZ] pi5: migration: /resourcelinks/3: 35 resources
```
You would enter the ID, `3`, in the Homebridge Hue configuration.

**Comment by kolorfilm** [comment]
Works like a charm! Thanks. I made a fresh Installation btw. The HomeKit wizard is really simple. I gave me for all devices a popup to already move the new fetched devices to existing groups.

I was wondering if there is a plan to bring back the adaptive live adjustments to the new version. For hue there is a option to make the lighting not too red.

I also recognised that when I turned on adaptive light for a group in a room like (lamp 1-3 grouped into a single lamp group) the brightness doesn’t set correctly (when I set the group to brightness 50 is lacks means doesn’t set up to 50 directly. just after a few tries). For single lamps this works for adaptive light. When adaptive light is off it works also for groups. I was wondering if this is more a Deconz problem itself. I mainly use ikea lamps btw.

**Comment by ebaauw** [comment]
Please stay on topic and don't pollute this issue with other questions.

**Comment by Knot3n** [comment]
What will happen with the automations from the exposed devices?
Do we need to "link" them again to the correct device/automation?

**Comment by ebaauw** [comment]
HomeKit sees the devices exposed by Homebridge deCONZ as different accessories as when exposed by Homebridge Hue.  When you no longer expose the accessories by Homebridge Hue, any associations with HomeKit rooms, groups, scenes, and automation are lost.  You need to change these to include the Homebridge deCONZ accessories instead.

**Comment by ebaauw** [comment]
That’s a good point: as long as you use auto discovery, Homebridge Hue will discover the deCONZ gateway and block Homebridge startup, until it has connected to it.  You can disable auto discovery by listing your Hue bridge(s) in config.json, under `hosts`, but that only works when the Hue bridges have a fixed IP address.

I probably need to cut a Homebridge Hue release that (optionally?) only auto discovers Hue bridges.

**Comment by Marfre888** [comment]
> **UPDATE AUG 2023** I've bumped Homebridge deCONZ to v1, as I think it's production ready. I concluded my migration to Homebridge deCONZ in May, and have been running Homebridge deCONZ exclusively for three months now, without any issues.
> 
> Consequently, Homebridge Hue support for deCONZ gateways has been deprecated, meaning you should start planning a migration to Homebridge deCONZ. For now, Homebridge Hue continues to expose devices connected to a deCONZ gateway, but I won't be adding support for new devices nor implement any feature requests not supported by the Hue bridge.
> 
> Note that Homebridge Hue2 is still just a placeholder, so continue to use Homebridge Hue to expose devices connected to a Hue bridge.

@ebaauw I noticed Homebridge-Hue has a migrate feature but I can't find any tutorials or wikis on how to do it. Is there any way to easily jump to Homebridge-Deconz without having to remove all the lights/switches from Homekit, I have so many that it will be a monumental task to set up all the scenes and automations again.

**Comment by stefansegers** [comment]
Hi guys, I am a real noob and got my Hoob4 yesterday. I tried to install the Philips Hue plugin but couldn't get it to work. After contacting support I got this link and and read this is going to be the new plugin for Philips Hue.

I tried to do the steps, but already stuck at step 2. Pairing is not working, whatever I try

Isn't there an easy way to get this to work?

**Comment by ebaauw** [comment]
Please don't hijack this issue.

**Comment by ebaauw** [comment]
v0.13.67 adds the `noDeconz` option, not to discover any deCONZ gateways.

**Comment by romanbalsiger** [comment]
> ### Familiarise yourself with Homebridge deCONZ
> 1. Install Homebridge deCONZ in a child bridge.  See Homebridge deCONZ README for details, incl. link how to setup a child bridge;
> 2. Pair Homebridge deCONZ to your deCONZ gateway; by default, it will expose all zigbee devices connected to the deCONZ gateway.
> 3. Temporarily create a new HomeKit Home, and pair the Homebridge deCONZ child bridge to that Home;
> 4. Check that your devices are exposed correctly.  Log an issue for devices that aren't exposed correctly or for which you need additional feature requests.
> 5. Play around with the dynamic configuration through the `ui` command-line tool, un-exposing and exposing devices, changing log level, how to expose devices, etc.
> 
> ### Move Homebridge deCONZ to production
> 7. Un-expose all devices in Homebridge deCONZ;
> 8. Remove Homebridge deCONZ from your test HomeKit home, and pair it to your real production HomeKit home.  Remove the test HomeKit home;
> 9. Expose a device (that passes the checks in step 4) in Homebridge deCONZ.  Note the message of the migration resource link in the Homebridge log.
>    You'll now see the accessory twice in HomeKit, once exposed by Homebridge Hue, and once by Homebridge deCONZ.
> 10. Configure the Homebridge deCONZ accessory in HomeKit: assigning it to the HomeKit rooms, groups, scenes, automations like the Homebridge Hue exposed accessory.
> 11. Update the Homebridge Hue config.json setting `homebridgeDeconz` to the ID of the migration resourcelink by Homebridge deCONZ from step 8.
> 12. Restart Homebridge Hue to apply the migration resourcelink, blacklisting the devices exposed by Homebridge deCONZ.
> 13. At your convenience, repeat steps 8, 9, 11 for each device connected to the deCONZ gateway.
> 
> ### Cleanup Homebridge Hue
> 14. After all devices have been migrated, remove the deCONZ gateway from the Homebridge Hue configuration.  When listing bridges and gateways using `hosts`, remove the IP address and port of the deCONZ gateway from the array.  When using automatic discovery, set `noDeconz`.
> 15. Uninstall Homebridge Hue once all gateways are migrated (unless you still use it for any Hue bridge).

Thank you very much for those instructions! BTW the BackUp function of the "Controller" App helped a lot

**Comment by ebaauw** [comment]
**UPDATE DEC 2023**
I've removed mention of the deCONZ gateway from the Homebridge Hue documentation and Wiki.  Homebridge Hue v0.13.70 is the last version that still supports the deCONZ gateway.   In januari, I will publish a v0.14 of Homebridge Hue, that will no longer support the deCONZ gateway.

As of v0.13.70, `ph` no longer supports the deCONZ gateway.  Use `deconz` to interact with the deCONZ API from the command line.  This utility is bundled with Homebridge deCONZ, or can be installed standalone from the [`hb-deconz-tools`](https://github.com/ebaauw/hb-deconz-tools) package.

In other words: plan your migration to Homebridge deCONZ now.  See https://github.com/ebaauw/homebridge-hue/issues/1070#issuecomment-1674780081 for details.

Note that Homebridge Hue2 is still a placeholder, so please continue to use Homebridge Hue for the gen-2 Hue bridge.

**Comment by Knot3n** [comment]
> ### Familiarise yourself with Homebridge deCONZ
> 
>     1. Install Homebridge deCONZ in a child bridge.  See Homebridge deCONZ README for details, incl. link how to setup a child bridge;
> 
>     2. Pair Homebridge deCONZ to your deCONZ gateway; by default, it will expose all zigbee devices connected to the deCONZ gateway.
> 
>     3. Temporarily create a new HomeKit Home, and pair the Homebridge deCONZ child bridge to that Home;
> 
>     4. Check that your devices are exposed correctly.  Log an issue for devices that aren't exposed correctly or for which you need additional feature requests.
> 
>     5. Play around with the dynamic configuration through the `ui` command-line tool, un-exposing and exposing devices, changing log level, how to expose devices, etc.
> 
> 
> ### Move Homebridge deCONZ to production
> 
>     7. Un-expose all devices in Homebridge deCONZ;
> 
>     8. Remove Homebridge deCONZ from your test HomeKit home, and pair it to your real production HomeKit home.  Remove the test HomeKit home;
> 
>     9. Expose a device (that passes the checks in step 4) in Homebridge deCONZ.  Note the message of the migration resource link in the Homebridge log.
>        You'll now see the accessory twice in HomeKit, once exposed by Homebridge Hue, and once by Homebridge deCONZ.
> 
>     10. Configure the Homebridge deCONZ accessory in HomeKit: assigning it to the HomeKit rooms, groups, scenes, automations like the Homebridge Hue exposed accessory.
> 
>     11. Update the Homebridge Hue config.json setting `homebridgeDeconz` to the ID of the migration resourcelink by Homebridge deCONZ from step 8.
> 
>     12. Restart Homebridge Hue to apply the migration resourcelink, blacklisting the devices exposed by Homebridge deCONZ.
> 
>     13. At your convenience, repeat steps 8, 9, 11 for each device connected to the deCONZ gateway.
> 
> 
> ### Cleanup Homebridge Hue
> 
>     14. After all devices have been migrated, remove the deCONZ gateway from the Homebridge Hue configuration.  When listing bridges and gateways using `hosts`, remove the IP address and port of the deCONZ gateway from the array.  When using automatic discovery, set `noDeconz`.
> 
>     15. Uninstall Homebridge Hue once all gateways are migrated (unless you still use it for any Hue bridge).

What will happen with automatins within Homekit? 
Will they break when i'll migrate all the devices?

**Comment by ebaauw** [comment]
> What will happen with automatins within Homekit?
Will they break when i'll migrate all the devices?

That's already been answered above.

**Comment by Knot3n** [comment]
> > What will happen with automatins within Homekit?
> > Will they break when i'll migrate all the devices?
> 
> That's already been answered above.

Sorry ... totally forgot that i already asked.. my bad. Christmas stress :=)

**Comment by spewu** [comment]
Sorry if this is considered spam, but I thought maybe my script could help someone else 🙂

I have a lot of devices, to calling the `ui` tool to un-expose each device is very tedious, so I created a small script to do it for me.

First get a json file with all of your devices:

```
ui get /devices > devices.json
```

Then create a simple script like this (just put this into a file called `un-expose-all.sh` or whatever you wanna call it):

```
#!/bin/bash

# Path to the JSON file
FILE="devices.json"

# Check if the JSON file exists
if [ ! -f "$FILE" ]; then
    echo "Error: File $FILE does not exist."
    exit 1
fi

# Read and process each device in the JSON file
jq -c 'to_entries[] | {id: .key, expose: false}' $FILE | while IFS= read -r line; do
    id=$(echo $line | jq -r '.id')
    # Call the 'ui' app with the updated device information
    ui put "/devices/$id" "{\"expose\": false}"
done

echo "Devices updated successfully."
```

Make the file executable:

```
chmod +x un-expose-all.sh
```

Then execute it:

```
./un-expose-all.sh
```

Once it has completed, all of your devices will be un-exposed.

**Comment by ebaauw** [comment]
See for https://github.com/ebaauw/homebridge-hue/wiki/Features comparison of HomeKit support of Hue bridge's native HomeKit feature vs Hue bridge's native Matter feature vs Hue bridge with Homebridge Hue vs deCONZ with Homebridge deCONZ.  It largely depends on the devices you have.  Next to HomeKit you might have other considerations like the Hue app, dependence on cloud services, or features like Hue Entertainment.

**Comment by hanwalt** [comment]
Hello ebaauw,
the hue addon wasn`t updated quite a time and hue2 is not ready, so my concern is that your hue plugin may not be compatible with upcoming Homebridge v2.0 version. Can you tell me if hue will be working with Homebridge v2.0?

**Comment by ebaauw** [comment]
I’m working, albeit slowly, on the next version to be Homebridge 2.0 ready.  It will probably end October, together with the update to NodeJS v22.  I’m removing any deCONZ-related features from the code, so I’ll have a clean base when starting on Homebridge Hue2.

**Comment by hanwalt** [comment]
Thanks for your information. I really appreciate your work for the best plugin of Homebridge!

**Comment by ebaauw** [comment]
**UPDATE NOV 2024**
Everyone should have migrated to Homebridge deCONZ by now.  With v0.14, Homebridge Hue no longer supports deCONZ gateways.

I cleaned up the code to remove any deCONZ-specific stuff, to have a clean base to start the migration to Homebridge Hue2.  Also, I've updated Homebridge Hue to support Homebridge v2 and NodeJS v22.  As of now, I won't be implementing any new features in Homebridge Hue, and focus on developing Homebridge Hue2.  
As that will be a modern plugin, using the dynamic platform accessories model (like Homebridge deCONZ) a migration will be needed from Homebridge Hue to Homebridge Hue2, similar to the migration to Homebridge deCONZ described above.

Homebridge Hue2 will re-use the underlying `hb-hue-tools` library, dealing with interacting with the Hue bridge and providing the `ph` utility.  This library now fully supports the Hue API v2, next to the Hue API v1 (try `ph get /resource`).

**Comment by bkarakashev** [comment]
@ebaauw Thanks for all your work. I wonder if there is any update on the Homebridge Hue 2?

**Comment by ebaauw** [comment]
Slowly.  I updated `hb-hue-tools` with full support for the Hue API v2 and the Hue bridge pro.  However, the Homebridge Hue2 plugin itself is still work in progress.

**Comment by DSimeone1989** [comment]
> Slowly. I updated `hb-hue-tools` with full support for the Hue API v2 and the Hue bridge pro. However, the Homebridge Hue2 plugin itself is still work in progress.

Dear Erik,
With Homebridge 2.0 available, can you give an update on your WiP on Hue v2?
Thank you very much! And your work is much appreciated!

Kind regards.

**Comment by ebaauw** [comment]
Homebridge Hue should work with Homebridge 2.0.  I'll publish a new version with updated dependencies shortly.

Homebridge Hue2 is, indeed, still work in progress.  I could publish a WiP version, but not sure if it will do anything useful yet.
